bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceHandler.java

```diff
@@ -0,0 +1,588 @@
+                    freeAtHomeBridge = (FreeAtHomeBridgeHandler) handler;
+
+                    device = freeAtHomeBridge.getFreeatHomeDeviceDescription(deviceID);
+                }
+            }
+        } catch (FreeAtHomeHttpCommunicationException e) {
+            logger.debug("Communication error during updateChannels - {}", e.getMessage());
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        for (int i = 0; i < device.getNumberOfChannels(); i++) {
+            FreeAtHomeDeviceChannel channel = device.getChannel(i);
+
+            for (int j = 0; j < channel.getNumberOfDatapointGroup(); j++) {
+                FreeAtHomeDatapointGroup dpg = channel.getDatapointGroup(j);
+
+                ChannelTypeUID channelTypeUID = UidUtils.generateChannelTypeUID(dpg.getValueType(), dpg.isReadOnly());
+
+                if (channelTypeProvider.getChannelType(channelTypeUID, null) == null) {
+                    channelTypeUID = createChannelTypeForDatapointgroup(dpg, channelTypeUID);
+                }
+
+                ChannelUID channelUID = UidUtils.generateChannelUID(thingUID, device.getDeviceId(),
+                        channel.getChannelId(), dpg.getLabel());
+
+                FreeAtHomeDatapoint outputDatapoint = dpg.getOutputDatapoint();
+
+                // in case of output channel, register it for updates
+                if (outputDatapoint != null) {
+                    // Register channel for event based state updated
+                    if (freeAtHomeBridge != null) {
+                        ChannelUpdateHandler updateHandler = freeAtHomeBridge.channelUpdateHandler;
+
+                        updateHandler.registerChannel(device.getDeviceId(), channel.getChannelId(),
+                                outputDatapoint.getDatapointId(), this, channelUID, dpg.getValueStateConverter());
+                    }
+                }
+
+                // add the datapoint group to the maping channel
+                mapChannelUID.put(channelUID, dpg);
+
+                logger.debug("Thing channelType reloaded - Device: {} - channelTypeUID: {}",
+                        device.getDeviceId() + device.getDeviceLabel(), channelTypeUID.getAsString());
+            }
+        }
+    }
+
+    public void removeChannels() {
+        FreeAtHomeBridgeHandler freeAtHomeBridge = null;
+
+        Bridge bridge = this.getBridge();
+
+        try {
+            if (bridge != null) {
+                ThingHandler handler = bridge.getHandler();
+
+                if (handler instanceof FreeAtHomeBridgeHandler) {
+                    freeAtHomeBridge = (FreeAtHomeBridgeHandler) handler;
+
+                    device = freeAtHomeBridge.getFreeatHomeDeviceDescription(deviceID);
+                }
+            }
+        } catch (FreeAtHomeHttpCommunicationException e) {
+            logger.debug("Communication error during updateChannels - {}", e.getMessage());
```

Comment:
This logging is not needed since the status update below will already be logged by the framework.